frontend/lib/parsing/ParserContextImpl.h

```diff
@@ -687,15 +677,15 @@ OpCall* ParserContext::buildUnaryOp(YYLTYPE location,
   // may reassign op here to match old parser
   // as in buildPreDecIncWarning
   if (ustrOp == "++") {
-    noteWarning(location, "++ is not a pre-increment");
+    CHPL_PARSER_REPORT(this, PreIncDecOp, location, "++", "increment");
```

Comment:
Since `++` determines `increment`, perhaps worth only passing one of them in, and computing the other in the error body. Make illegal calls impossible, eh?